__main__.py

```diff
@@ -3,81 +3,149 @@
 import datetime as dt
 import json
 import sys
+from enum import Enum
 from pathlib import Path
-from typing import NoReturn
+from typing import TYPE_CHECKING, Any, ClassVar, NewType, NoReturn, TypedDict, cast
 
 import requests
 from dahlia import dahlia, dprint
+from requests.exceptions import HTTPError
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
```

Comment:
Am importing `typing_extensions.Self` from inside this `if TYPE_CHECKING` block so it doesn't become a runtime dependency.